### DISCUSSION TITLE
Revalidation with `revalidateTag` not working when app is deployed

### DISCUSSION THREAD

**Discussion by HristoHentov** [vercel/next.js#59462] [Help]
## Problem
Using `revalidateTag` works locally with `npm run build && npm run start` but stops working, once app is deployed in a cloud environment

## Details
* I am using `"next": "14.0.4-canary.41"`, on node `20.9.0`
* Most of my pages get their content from my headless CMS via fetch, with the following call 
`const res = await fetch(fullUrl, { next: { tags: ['cms']} });`

* I have implemented a `POST` request handler, which calls `revalidateTag('cms');` to invalidate all my cached data.

## Problem 
* Running my app locally with `npm run build && npm run start`, performs as expected. 
Anytime a `POST` request is sent to the route for my requestHandler, all content is invalidated, and any changes are reflected in my final rendered HTML. 
**However**, when deploying my app (on Digital Ocean's App Platform), calling my request handler returns a 200 success content, but my content remains stale, event after multiple page refreshes.

* As my issue cannot be reproduced locally, I am looking for any hints as to what might be going on. Also if there are any debug-level NextJs logs which I can enable to see what exactly is happening, that would also be helpful

## Attempts to resolve the issue
* I tried various versions of NextJS (I cannot go beyond `14.0.4-canary.41` as 1`4.0.4-canary.42` includes [59254](https://github.com/vercel/next.js/pull/59254) which introduces a bug when using `@mui/material`. 

* I tried running the revalidateTag in a `setTimeout` after a couple of seconds, to avoid any potential race conditions, but this did not have an effect.

* I have verified my endpoint is getting called, and not throwing any errors by including `console.log()` statemnts + verifying the response.

* I assume my setup is correct as running the app locally performs exactly as expected.

### Additional information

```js
Local `next info`

Operating System:        
  Platform: win32        
  Arch: x64
  Version: Windows 10 Pro
Binaries:
  Node: 20.9.0
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.0.4-canary.41
  eslint-config-next: 14.0.4
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.3.2
Next.js Config:
  output: N/A
```

Digital Ocean's deployment is running `Ubuntu 22.04` and running node `v20.10.0`
```


### Example

No minimal reproduction available, as the issue is only happening when I deploy my application

**Comment by XoL1507** [comment]
there is a pro that i have to pay.. i msg the sales no reply. and seriously i dont have money..
